lib/dencode_context.js

```diff
@@ -0,0 +1,214 @@
+'use strict'; // eslint-disable-line strict
+
+const assert = require('assert');
```

Comment:
Pedantic: [shan't we use import now](https://github.com/airbnb/javascript/blob/master/README.md#modules) ?

Reply:
Didn't find a trace in the official nodejs docs, so no. And it seems not to do what we want here: using import will load the module asynchronously (https://medium.com/the-node-js-collection/an-update-on-es6-modules-in-node-js-42c958b890c).